prometric-derive/src/expand.rs

```diff
@@ -452,6 +455,17 @@ pub fn expand(metrics_attr: MetricsAttr, input: &mut ItemStruct) -> Result<Token
         #input
     };
 
+    let static_decl = if metrics_attr.with_static {
+        let static_name = format_ident!("{}", to_screaming_snake(&ident.to_string()));
+        Some(quote! {
+            /// A static instance of the metrics, initialized with default values.
+            /// This static is generated when `with_static` is enabled on the `#[metrics]` attribute.
+            #vis static #static_name: std::sync::LazyLock<#ident> = std::sync::LazyLock::new(|| #ident::default());
```

Comment:
We're using `default()` here (which currently seems like the only option). This means that the builder here cannot be used, at least with `LazyLock`. Checking if there's an alternative here so we can still allow custom registries and static labels.